### ISSUE TITLE
Reverse-proxying

### ISSUE THREAD

**Issue by makeworld-the-better-one** [pitr/gig#1]
Feature request.

Gig looks great, nice work! I was wondering if reverse-proxying is on your roadmap, so all requests under a certain route or whatever is sent to a different Gemini server.

**Comment by pitr** [comment]
this can be implemented using `Context.Stream()` or `Context.Response()` (which implements `io.Writer` interface), and using match-all routes, eg. `g.Handle("/proxy/*",..)`. But a builtin middleware would be nicer if it would be contributed :)

**Comment by makeworld-the-better-one** [comment]
As Michael noted in [this comment](https://github.com/michael-lazar/jetforce/issues/36#issuecomment-657164690) this would break client certificates, but there might be other solutions like SNI proxies. Thanks for the info though.

**Comment by pitr** [comment]
I thought you meant more of an tls-terminating proxy where server makes request on behalf of client

**Comment by makeworld-the-better-one** [comment]
Yes, that is what I was thinking originally. I only brought up SNI proxies because Michael did, and I think it's an interesting solution. The end result in most cases would be the same, better even, because client certs would remain intact.
